[
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:35:28 Incompatible variable type [9]: docker_repo is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:58:29 Incompatible variable type [9]: docker_repo is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:94:12 Incompatible variable type [9]: docker_repo is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:213:12 Incompatible variable type [9]: password is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:214:12 Incompatible variable type [9]: ssh_key is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:215:12 Incompatible variable type [9]: docker_repo is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:300:12 Incompatible variable type [9]: password is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:301:12 Incompatible variable type [9]: ssh_key is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:322:8 Incompatible return type [7]: Expected `str` but got `typing.Tuple[typing.Optional[str], typing.Any]`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:477:12 Incompatible variable type [9]: username is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:478:12 Incompatible variable type [9]: webui is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:479:12 Incompatible variable type [9]: jobui is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/clusterlib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/clusterlib.py:480:12 Incompatible variable type [9]: jupyter is declared to have type `str` but is used as type `None`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/clusterlib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/core/command_builder.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/core/command_builder.py:19:36 Incompatible variable type [9]: value is declared to have type `str` but is used as type `None`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/core/command_builder.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/core/command_builder.py:19:55 Incompatible variable type [9]: enable is declared to have type `bool` but is used as type `None`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/core/command_builder.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/core/command_builder.py:32:12 Incompatible variable type [9]: enable is declared to have type `bool` but is used as type `str`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/core/ssh.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk_sdk/utils/get_ssh_key.py",
    "file_hunks_size": 1,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "aztk/core/ssh.py:4:24 Incompatible variable type [9]: key_or_path is declared to have type `str` but is used as type `None`.",
    "message": " key_or_path is declared to have type `str` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 4,
    "warning_line": "def get_user_public_key(key_or_path: str = None, secrets_config: SecretsConfig = None):"
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/core/ssh.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk_sdk/utils/get_ssh_key.py",
    "file_hunks_size": 1,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "aztk/core/ssh.py:4:49 Incompatible variable type [9]: secrets_config is declared to have type `SecretsConfig` but is used as type `None`.",
    "message": " secrets_config is declared to have type `SecretsConfig` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 4,
    "warning_line": "def get_user_public_key(key_or_path: str = None, secrets_config: SecretsConfig = None):"
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/error.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/error.py:7:23 Incompatible variable type [9]: message is declared to have type `str` but is used as type `None`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/error.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/error.py:13:23 Incompatible variable type [9]: message is declared to have type `str` but is used as type `None`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/joblib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/joblib.py:54:46 Incompatible parameter type [6]: Expected `str` for 2nd positional only parameter to call `aztk.core.command_builder.CommandBuilder.add_option` but got `typing.Union[List[str], str]`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/joblib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/joblib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/joblib.py:55:50 Incompatible parameter type [6]: Expected `str` for 2nd positional only parameter to call `aztk.core.command_builder.CommandBuilder.add_option` but got `typing.Union[List[str], str]`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/joblib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/joblib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/joblib.py:56:46 Incompatible parameter type [6]: Expected `str` for 2nd positional only parameter to call `aztk.core.command_builder.CommandBuilder.add_option` but got `typing.Union[List[str], str]`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/joblib.py'",
    "dd_fail": true
  },
  {
    "project": "Azure/aztk",
    "commit": "e31d5e73aee754655b734d51e1bad4068fdc9585",
    "filename": "aztk/joblib.py",
    "min_patch_found": false,
    "full_warning_msg": "aztk/joblib.py:126:12 Incompatible parameter type [6]: Expected `str` for 4th parameter `app_args` to call `Job.app_submit_cmd` but got `List[str]`.",
    "exception": "[Errno 2] No such file or directory: '/home/chowyi/TypeAnnotation_Study/GitHub/Azure-aztk/aztk/joblib.py'",
    "dd_fail": true
  }
]